app/models/geographic_item.rb

```diff
@@ -121,91 +138,165 @@ class GeographicItem < ApplicationRecord
+
+        GeographicItem.select(
+          'ST_Buffer(' \
+            "#{GeographicItem::GEOGRAPHY_SQL}#{geometric}, " \
+            "#{buffer}" \
+          ') AS buffer'
+        )
+        .where(id: geographic_item_id)
+        .first.buffer
       end
 
-      # @return [GeographicItem::ActiveRecord_Relation]
-      # @params [Array] array of geographic area ids
-      def default_by_geographic_area_ids(geographic_area_ids = [])
-        GeographicItem.
-          joins(:geographic_areas_geographic_items).
-          merge(::GeographicAreasGeographicItem.default_geographic_item_data).
-          where(geographic_areas_geographic_items: {geographic_area_id: geographic_area_ids})
+      def st_distance_sql(shape_sql)
+        'ST_Distance(' \
+          "#{GeographicItem::GEOGRAPHY_SQL}, " \
+          "(#{shape_sql})" \
+        ')'
+      end
+
+      def st_area_sql(shape_sql)
+        'ST_Area(' \
+          "(#{shape_sql})" \
+        ')'
+      end
+
+      def st_isvalid_sql(shape_sql)
+        'ST_IsValid(' \
+          "(#{shape_sql})" \
+        ')'
+      end
+
+
+      def st_isvalidreason_sql(shape_sql)
+        'ST_IsValidReason(' \
+          "(#{shape_sql})" \
+        ')'
+      end
+
+      def st_astext_sql(shape_sql)
+        'ST_AsText(' \
+          "(#{shape_sql})" \
+        ')'
+      end
+
+      def st_minimumboundingradius_sql(shape_sql)
+        'ST_MinimumBoundingRadius(' \
+          "(#{shape_sql})" \
+        ')'
+      end
+
+      # True for those shapes that are within `distance` of (i.e. intersect the
+      # `distance`-buffer of) the shape_sql shape. This is a geography dwithin,
+      # distance is in meters.
+      def st_dwithin_sql(shape_sql, distance)
+        'ST_DWithin(' \
+            "(#{GeographicItem::GEOGRAPHY_SQL}), " \
+            "(#{shape_sql}), " \
+            "#{distance}" \
+          ')'
+      end
+
+      def st_asgeojson
+        select("ST_AsGeoJSON(#{GeographicItem::GEOMETRY_SQL.to_sql}) geo_json")
       end
 
       # @param [String] wkt
       # @return [Boolean]
-      #   whether or not the wtk intersects with the anti-meridian
+      #   whether or not the wkt intersects with the anti-meridian
       #   !! StrongParams security considerations
       def crosses_anti_meridian?(wkt)
         GeographicItem.find_by_sql(
           ['SELECT ST_Intersects(ST_GeogFromText(?), ST_GeogFromText(?)) as r;', wkt, ANTI_MERIDIAN]
         ).first.r
       end
 
-      # @param [Integer] ids
-      # @return [Boolean]
-      #   whether or not any GeographicItem passed intersects the anti-meridian
-      #   !! StrongParams security considerations
-      #   This is our first line of defense against queries that define multiple shapes, one or
-      #   more of which crosses the anti-meridian.  In this case the current TW strategy within the
-      #   UI is to abandon the search, and prompt the user to refactor the query.
-      def crosses_anti_meridian_by_id?(*ids)
```

Comment:
Not sure if we are going to replace this elsewhere- is still a useful concept to maintin (counter to my "Rip it out").